### DISCUSSION TITLE
Microsoft Graph Multitenant

### DISCUSSION THREAD

**Discussion by adriantheking** [AzureCommunityPL/group#222] [Pomoc] [answered]
Cześć,

Mam następujący scenariusz:
Tenant1:
Hostowana jest webowa aplikacja administracyjna
Tenant2: 
Hostowana jest webowa aplikacja końcowa. Dodatkowo jako resource stworzone zostało tutaj ADB2C z użytkownikami którzy posiadają dostęp do aplikacji końcowej. 

Z poziomu aplikacji webowej administracyjnej chciałbym mieć możliwość administrowania przez Microsoft Graph użytkownikami z ADB2C. Udało mi się to osiągnąć przez aplikację stworzoną w ADB2C i uwierzytelnienie niej przez ClientSecretCredentials jednak wymaga ona podania również ClientSecret. 
Czy jest jakaś inna możliwość osiągnięcia tego ograniczając się do podania jedynie  TenantId oraz ClientId  bez secret ?

**Comment by PawelHaracz** [comment, marked as answer]
Hej,

Uzyj w aplikacji administracyjnej on Behalf on flwo

https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow

Uwierzytelniony user uderza do Twojej apki, i na podstawie tego generujesz kolejny token z innej aplikacji która ma już uprawnienia do graph api (clientId i clientSecret są trzymane po stronie serwera) i pobierasz dane z graphApi
